mlflow/johnsnowlabs.py

```diff
@@ -0,0 +1,874 @@
+        out_folder)
+
+
+HEALTHCARE_wheel_URI = 'https://pypi.johnsnowlabs.com/{secret}/spark-nlp-jsl/spark_nlp_jsl-{SPARK_NLP_INTERNAL_VERSION}-py3-none-any.whl'
+_JOHNSNOWLABS_MODEL_PATH_SUB = "jsl-model"
+_MLFLOWDBFS_SCHEME = "mlflowdbfs"
+
+_logger = logging.getLogger(__name__)
+
+
+def get_default_pip_requirements():
+    """
+    :return: A list of default pip requirements for MLflow Models produced by this flavor.
+             Calls to :func:`save_model()` and :func:`log_model()` produce a pip environment
+             that, at minimum, contains these requirements.
+    """
+    if not 'SECRET' in os.environ:
+        raise Exception(
+            'Please set SECRET environment variable to your secret key before saving or logging your model')
+    from johnsnowlabs import settings
+    return [
+        f'johnsnowlabs=={settings.raw_version_jsl_lib}',
+        f"https://pypi.johnsnowlabs.com/{os.environ['SECRET']}/spark-nlp-jsl/spark_nlp_jsl-{settings.raw_version_medical}-py3-none-any.whl",
+        f'pyspark=={settings.raw_version_pyspark}',
+        'mlflow-tmp==2.2.26', # TODO REMOVE THIS after PR for johnsnowlabs flavor is merged
+        'pandas==1.5.3',
+        # 'mleap',  # TODO JUST ADD TO TEST CLASS?
+        # 'pytest',  # TODO JUST ADD TO TEST CLASS?
+
+    ]
+
+
+def get_default_conda_env():
+    """
+    :return: The default Conda environment for MLflow Models produced by calls to
+             :func:`save_model()` and :func:`log_model()`. This Conda environment
+             contains the current version of PySpark that is installed on the caller's
+             system. ``dev`` versions of PySpark are replaced with stable versions in
+             the resulting Conda environment (e.g., if you are running PySpark version
+             ``2.4.5.dev0``, invoking this method produces a Conda environment with a
+             dependency on PySpark version ``2.4.5``).
+    """
+    return _mlflow_conda_env(additional_pip_deps=get_default_pip_requirements())
+
+
+@format_docstring(LOG_MODEL_PARAM_DOCS.format(package_name="johnsnowlabs"))
+def log_model(
+        spark_model,
+        artifact_path,
+        conda_env=None,
+        code_paths=None,
+        dfs_tmpdir=None,
+        sample_input=None,
+        registered_model_name=None,
+        signature: ModelSignature = None,
+        input_example: ModelInputExample = None,
+        await_registration_for=DEFAULT_AWAIT_MAX_SLEEP_SECONDS,
+        pip_requirements=None,
+        extra_pip_requirements=None,
+        metadata=None,
+        ### Secret Params
+
+):
+    """
+    Log a Johnsnowlabs NLUPipeline, created via nlp.load() model as an MLflow artifact for the current run. This uses the
+    MLlib persistence format and produces an MLflow Model with the Johnsnowlabs flavor.
+
+    Note: If no run is active, it will instantiate a run to obtain a run_id.
+
+    :param spark_model: Spark model to be saved - MLflow can only save descendants of
```

Comment:
Should this docstring entry be updated to reflect the nature of this flavor?

Reply:
Yes thanks, updated accordingly in latest commit